### DISCUSSION TITLE
Ability to add new required arguments

### DISCUSSION THREAD

**Discussion by brandonkelly** [rectorphp/rector#8242] [Ideas]
`Rector\Arguments\ValueObject\ArgumentAdder` requires that you specify a default value for the argument.

Related to #8240, it would be nice if there was a `RequiredArgumentAdder` alternative that could be used to add a new argument _without_ a default value.

```php
$rectorConfig
    ->ruleWithConfiguration(ArgumentAdderRector::class, [
        new RequiredArgumentAdder('SomeExampleClass', 'someMethod', 0, 'someArgument', new ObjectType('SomeType')),
    ]);
```

```diff
 $someObject = new SomeExampleClass;
 $someObject->someMethod();

 class MyCustomClass extends SomeExampleClass
 {
-    public function someMethod()
+    public function someMethod(SomeType $someArgument)
     {
     }
 }
```

**Comment by staabm** [comment]
Adding a new parameter to a method signature itself does not seem very useful. 

The harder question would be, to fix all the call-sites. since you are adding a new non defaulting param, you need to adjust every call.

Without more context and constraints this seems undoable from a logic point of view

**Reply by brandonkelly** [reply]
`ArgumentAdderRector` already exists, FWIW. I’m just asking for the ability to omit the default value.

My use case is [craftcms/rector](https://github.com/craftcms/rector), which helps Craft CMS plugins with major Craft version upgrades. Craft 5.0 adds a couple new arguments to existing methods that plugins could be extending. The base methods’ arguments don’t have default values, so it’s weird that extended methods created via Rector ([here](https://github.com/craftcms/rector/blob/cb9d3d817a36dc9ea526524a46e22d6d70ccd3e9/sets/craft-cms-50.php#L108-L114)) will have default arguments that aren’t necessary.
